### PR TITLE
docs(*) revamp and reorganize documentation

### PR DESCRIPTION
**Pull request by hbagdi**
This is part of a patchset aimed to improve
and add documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,99 +51,105 @@ The following matrix lists supported versions of Kong for every release of Kong
...
-
-## How it works?
-
-Using a standard [Kubernetes deployment][10], the Ingress controller runs multiple containers in a single pod. This allows us to define in a unit containing an [initContainer][11] to run Kong migrations, one container for the Kong admin API in control-plane mode and one container for the ingress controller itself. With this approach we simplify the deploy of the required components without user intervention.
-Once the deployment passes the readiness and liveness probes it means the Kong migrations ran and ingress controller can communicate with the Kong admin API.
-
-In a different deployment we run Kong in data-plane mode. This means the Kong instances only expose the proxy ports.
-By using this approach we can deploy and scale the data-plane with the requirements of your applications, i.e. using a daemonset, a deployment with affinity rules, etc.
+You can deploy Kong Ingress Controller on any
+Kubernetes cluster which Service of `type: LoadBalancer`.
```
Comment (coopr) [inline README.md]: "which" should probably be "with" but I'm not certain

File: README.md [hunk 2]
```diff
@@ -19,99 +51,105 @@ The following matrix lists supported versions of Kong for every release of Kong
...
 
-The next diagram shows how the components interact:
+You can use
+[Minikube](https://kubernetes.io/docs/setup/minikube/)
+on your local machine or use
+a hosted k8s service like
+[GKE](https://cloud.google.com/kubernetes-engine/).
```
Comment (coopr) [inline README.md]: Now that it is GA, I'd suggest you also mention Docker Desktop https://www.docker.com/products/docker-desktop - it makes it fast and easy to get started with Kubernetes.

File: README.md [hunk 3]
```diff
@@ -19,99 +51,105 @@ The following matrix lists supported versions of Kong for every release of Kong
...
-metadata:
-  name: foo
-  annotations:
-    kubernetes.io/ingress.class: "gce"
-```
+Please search through the posts on
+[Kong Nation](https://discuss.konghq.com/c/kubernetes) as it's
+likely that another user has run into the same problem.
+If you don't find an answer, please feel free to post a question.
+If you've a feature request, please post on in
```
Comment (coopr) [inline README.md]: using "you've" in this context is a little unusual - I'd suggest the full "If you have a..."

Comment (coopr) [reply README.md]: "...please post on in..." -> "...please post in..."

File: README.md [hunk 4]
```diff
@@ -19,99 +51,105 @@ The following matrix lists supported versions of Kong for every release of Kong
...
+[Feature Suggestions](https://discuss.konghq.com/c/feature-suggestions)
```
Comment (coopr) [inline README.md]: Do we want feature suggestions for the ingress controller in that category, or in the Kubernetes category? I'm not sure :)

File: README.md (hunk 2 above)
Comment (hbagdi) [reply README.md]: This is new!
We've not tested or worked with it yet, so we can't include it yet as we won't be able to solve support issues with it.
Once we're familiar and have verified that it works well with our Ingress Controller, we sure can include it!

File: README.md (hunk 4 above)
Comment (hbagdi) [reply README.md]: Yeah, I wasn't sure either so went ahead with one.

**Pull request merged**